pkg/k8s/watchers/watcher.go

```diff
@@ -296,7 +296,18 @@ func resourceGroups(cfg WatcherConfiguration) (resourceGroups, waitForCachesOnly
 // already been registered.
 func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context, cachesSynced chan struct{}) {
 	resources, cachesOnly := k.resourceGroupsFn(k.cfg)
+	k._initK8sSubsystem(ctx, cachesSynced, resources, cachesOnly)
+}
+
+// InitK8sSubsystemWithResources takes a channel for which it will be closed when all
+// the resources are synchronized. Different from InitK8sSubsystem, this function
+// allows watcher to only sync the resources that are passed in.
+func (k *K8sWatcher) InitK8sSubsystemWithResources(ctx context.Context, cachesSynced chan struct{}, resources []string) {
```

Comment:
What's the plan for consuming these changes? We usually only introduce changes into the tree along with the code that uses it, to avoid bloating the codebase with unused functions. Unless the user is in-tree, there will be no way for Cilium contributors or maintainers to understand the repercussions of this, and at some point someone will likely notice that this function is unused, then delete it again.

Reply:
@joestringer I found a better way to do this, without introducing unused code in the tree:
https://github.com/cilium/cilium/pull/37886